File: Devansh_Sharma_2301220100065/0069-sqrtx/0069-sqrtx.py
class Solution:
    def mySqrt(self, x: int) -> int:
        # for i in range(1,x+1):
        #     if i*i>x:
        #         return i-1
        # return x
        if x==0:
            return x
        l,r=0,x
        while l<=r:
            m=l+(r-l)//2
            if m*m==x:
                return m
            elif m*m>x:
                r=m-1
            else:
                l=m+1
        return r

File: 2301221530069_Yash Asthana_DSA/DSA-Yash-asthana-2301221530069/0069-sqrtx/0069-sqrtx.java
class Solution {
    public int mySqrt(int x) {
        long start = 0;
        long end = x;
        long ans = 0;

        while (start <= end) {
            long mid = start + (end - start) / 2;

            if (mid * mid == x) {
                return (int) mid;
            }

            if (mid * mid > x) {
                end = mid - 1;
            } else {
                ans = mid;
                start = mid + 1;
            }
        }

        return (int) ans;
    }
}